lib/phoenix_live_view/channel.ex

```diff
@@ -837,6 +841,11 @@ defmodule Phoenix.LiveView.Channel do
     end
   end
 
+  defp push_pending_events_on_redirect(state, socket) do
```

Comment:
Consider moving this function to the Diff module.

Reply:
I think the function makes sense here because it depends on `push` maybe we could add a `get_push_events` function to `Diff` which returns either {"diff", %{e: events}} or `nil`?

Presumably the goal here is to avoid the duplication of "diff" and `@events`?

Reply:
Yeah. get_push_events sounds good to me!

Reply:
@josevalim how's this? https://github.com/phoenixframework/phoenix_live_view/compare/4cf29a0a4c474fc3556c1ac61db9be92bf9449f0..8aa91ace9d901a93cd7103a60aa70016354f515b